src/metabase/actions.clj

```diff
@@ -145,39 +146,57 @@
 
 ;;;; Action definitions.
 
-;;; Common base spec for all Actions. All Actions at least require
+;;; Common base spec for *all* Actions. All Actions at least require
 ;;;
-;;;    {:database <id>, :query {:source-table <id>}}
+;;;    {:database <id>}
+;;;
+;;; Anything else required depends on the action type.
 
 (s/def :actions.args/id
   (s/and integer? pos?))
 
-(s/def :actions.args.crud.common/database
+(s/def :actions.args.common/database
```

Comment:
I refactored this stuff a little bit more. `:database` is now a required key for all actions that use `perform-action!`. `{:query {:source-table <id>}}` is not since it's only used by the existing `:row/*` actions.

Reply:
I tweaked the spec names a bit to reflect this